hypothesis-python/src/hypothesis/strategies/_internal/core.py

```diff
@@ -120,6 +120,34 @@
     Protocol = object  # type: ignore[assignment]
 
 
+try:
+    import typing_extensions
+except ImportError:
+    typing_extensions = None  # type: ignore
+
+
+RequiredTypes: tuple = ()
+try:
```

Comment:
Let's keep all this logic in `types.py` with the other cases, and then import it into `core.py`.

Reply:
ok